src/redux/books/bookSlice.js

```diff
@@ -1,45 +1,73 @@
-import { createSlice } from '@reduxjs/toolkit';
-import { v4 as uuidv4 } from 'uuid';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
 
 const initialState = {
-  numOfBooks:
-[
-  {
-    id: uuidv4(),
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    id: uuidv4(),
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    id: uuidv4(),
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-],
+  books: [],
+  loading: false,
+  error: '',
```

Comment:
- [ ] I would suggest you to make sure that an action creator called `fetchBooks` exists in the `booksSlice` so as your the component will rendered after Fetch your books from the Bookstore API
as this will refactor your add book and remove book features to persist your changes in the server
````javascript
// use this line of code as an example, you can create this method in different ways.

import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
import axios from "axios"

export const fetchBooks = createAsyncThunk("books/fetchBooks", async () => {
  const response = await axios.get(
    "https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/"
  )
  return response.data
})
````